### PR TITLE
Restructuring Dependency

### PR DESCRIPTION
**Pull request by Geosynopsis**
The ARCSI had older dependencies on GDAL and RSGISLib. Dependencies updated to comply with those.

### CODE REVIEW

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
Seems `py6s` is pinned to `pysolar` 0.6 which isn't available for Python 3.7. `pysolar` 0.8 is though...

**Comment by Geosynopsis** [comment]
> Seems `py6s` is pinned to `pysolar` 0.6 which isn't available for Python 3.7. `pysolar` 0.8 is though...


Do you suggest raising issue there?

**Comment by petebunting** [comment]
This is a py6S issue is doesn’t work with pysolar > 0.7. I’ll try and have a look at what changed with pysolar release 0.7 and what is used in py6s. For the py6s functions used in ARCSI pysolar is probably not used at all I would think.

**Comment by Geosynopsis** [comment]
Shall we render the arcsi-feedstock for py2.7, py3.5 and py3.6 only for the moment?

**Comment by petebunting** [comment]
That would be best, I’ll try and look at the pysolar issue over the next month.

**Comment by Geosynopsis** [comment]
@conda-forge-admin, please rerender

**Comment by Geosynopsis** [comment]
As the new conda-forge handles the pinning in different fashion, @gillins updated the recipe. However, if we take `host: python` smithy builds and tests for all python matrix 2.7, 3.5, 3.6 and 3.7. It ignores `max_py_ver` option in `conda-forge.yml` . Therefore, the repository was build exclusively with attached `conda_build_config.yml` file as follows:

`conda smithy rerender -e conda_build_config.yml`

However, the exclusive config file misses to generate following key-value pairs in `.ci_support/win_python3.6.yaml`:

```
build_number_decrement:
  - '0'
channel_sources:
  - conda-forge,defaults
channel_targets:
  - conda-forge main
```
I will raise an issue in conda-smithy for this behavior.

**Comment by gillins** [comment]
@Geosynopsis I think you are meant to update the `skip` section in `meta.yaml` and then re-render to get the right behaviour rather than editing the CI files directly....

**Comment by Geosynopsis** [comment]
> @Geosynopsis I think you are meant to update the `skip` section in `meta.yaml` and then re-render to get the right behaviour rather than editing the CI files directly....

Do you meant for generating ci files for py35 also or in general for rerendering ? When I rerender without the exclusive file, smithy rerenders for all python matrices 2.7, 3.6, 3.7 which is now default in [conda-forge](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml). Is there a way to use exclusive file while using conda web admin?

**Comment by Geosynopsis** [comment]
@conda-forge-admin, please rerender

**Comment by Geosynopsis** [comment]
@petebunting @gillins Does it make sense to have explicit python <=3.6 in the requirements ? As it seems with that the tests only uses the last python version satisfying this criteria, not all python versions in the matrix as before.

**Comment by petebunting** [comment]
@Geosynopsis <https://github.com/Geosynopsis> thanks for your efforts here. While the dependencies aren’t available for >3.6 that seems like a pragmatic solution. Does conda-forge honour the requirements in the build process or will it just mean the 3.7 build fails with a requirements error?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,35 +10,24 @@ source:
   sha256: 5a53e5e9b1769c41d28e2a6644d4add68c1c922d93a882bd69fa457a47cfb5d5
 
 build:
-  number: 1
+  number: 2
   # Windows recipe includes .bat file copy so use bld.bat
   script: python setup.py install  # [unix]
   # rsgislib not available due to mpfr only supporting vc14 and above
   skip: True  # [win and py<35]
```
Comment (gillins) [inline recipe/meta.yaml]: This should read:
```
skip: True  # [(win and py<35) or (py>36)]
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,35 +10,24 @@ source:
...
-    - python
-    - vc 9  # [win and py27]
-    - vc 10  # [win and py34]
-    - vc 14  # [win and (py35 or py36)]
-
+  host:
+    - python <=3.6
   run:
-    - python
+    - python <=3.6
```
Comment (gillins) [inline recipe/meta.yaml]: Just leave this as `python`

File: recipe/meta.yaml (hunk 1 above)
Comment (gillins) [reply recipe/meta.yaml]: Then re-render....

**Comment by gillins** [comment]
> Does conda-forge honour the requirements in the build process

The re-rendering looks the `skip` line in `meta.yaml` to determine which versions of Python/platforms to build for.

File: recipe/meta.yaml (hunk 2 above)
Comment (Geosynopsis) [reply recipe/meta.yaml]: > > Does conda-forge honour the requirements in the build process
> 
> The re-rendering looks the `skip` line in `meta.yaml` to determine which versions of Python/platforms to build for.

Thank on this info. I was unaware of such setup.

**Comment by Geosynopsis** [comment]
@conda-forge-admin, please rerender

**Pull request merged**